plugins/by-name/openscad/default.nix

```diff
@@ -53,10 +69,9 @@ in
     in
     mkIf cfg.enable {
       extraPlugins =
-        with pkgs.vimPlugins;
         [ cfg.package ]
-        ++ (optional (fuzzyFinder == "skim") skim-vim)
-        ++ (optional (fuzzyFinder == "fzf") fzf-vim);
+        ++ (optional (fuzzyFinder == "skim" && cfg.skimPackage != null) cfg.skimPackage)
+        ++ (optional (fuzzyFinder == "fzf" && cfg.fzfPackage != null) cfg.fzfPackage);
```

Comment:
I wonder if gating installation behind _both_ a nullable package option _and_ another option makes sense, especially elsewhere where we use `hasInfix` on a settings option... :thinking: 

Maybe we should just provide a `fyzzyFinder` package option and if it isn't null, install it?

I'm not sure what the current enum is used for elsewhere, but perhaps it could default to the package's `meta.mainProgram`?

Reply:
I liked the idea and went through and implemented it... but, it looks like these are vimPlugins which don't get a meta.mainProgram. 
```bash
 error: attribute 'mainProgram' missing
       at /nix/store/2bmb1607gm9rfsm39lx6wdz45a9qdbzs-source/plugins/by-name/openscad/default.nix:68:35:
           67|         (mkIf (cfg.fuzzyFinderPackage != null) {
           68|           openscad_fuzzy_finder = cfg.fuzzyFinderPackage.meta.mainProgram;
             |                                   ^
           69|         })
```

Reply:
Thanks for trying. I'm guessing we also can't reliably use `lib.getName` either, since the package name may not match the exe name.

Maybe we could have something like:

```nix
fuzzyFinder = lib.mkOption {
  type = nullOr str; # or enum still
};
fuzzyFinderPackage =
  let
    defaults = {
      skim = "skim-vim";
      fzf = "fzf-vim";
    };
  in
    lib.mkOption {
      type = nullOr package;
      default =
        if defaults ? cfg.fuzzyFinder then
          getAttrByPath (toList defaults.${cfg.fuzzyFinder}) null pkgs
        else
          null;
      defaultText = literalMD ''
        # TODO
      '';
    };
```

IDK if there's a good way to generate the default text from the defaults attrs. I don't think it'd be easy to use `mkPackageOption` if we need such custom logic?

Reply:
Yeah, that's what was sounding like too much of a hassle. It's either something we need to maintain the available options explicitly through multiple package options or through that lookup function, basically. Do you think we just rip the migration out of this package ?

Reply:
I'm gonna pull it out of this PR, for now. This module is in need of a refactor and can be handled then.